server/application-server/src/main/java/de/tum/cit/aet/helios/github/GitHubClientManager.java

```diff
@@ -0,0 +1,321 @@
+package de.tum.cit.aet.helios.github;
+
+import de.tum.cit.aet.helios.util.GitHubAppJwtHelper;
+import java.io.File;
+import java.io.IOException;
+import java.security.PrivateKey;
+import java.time.Instant;
+import java.util.List;
+import java.util.concurrent.locks.ReentrantLock;
+import lombok.Getter;
+import lombok.extern.log4j.Log4j2;
+import okhttp3.OkHttpClient;
+import org.kohsuke.github.GHApp;
+import org.kohsuke.github.GHAppInstallation;
+import org.kohsuke.github.GHAppInstallationToken;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.github.GitHubBuilder;
+import org.kohsuke.github.extras.okhttp3.OkHttpGitHubConnector;
+import org.springframework.beans.factory.annotation.Value;
+
+/**
+ * Manages the GitHub client for the application.
+ *
+ * <p>This class is responsible for creating and refreshing the GitHub client
+ * used to interact with the GitHub API. It handles authentication using
+ * either a GitHub App or a personal access token (PAT), and ensures that
+ * the client is always up-to-date with a valid token.
+ * </p>
+ *
+ * <p>The GitHub client is cached and refreshed every 20 minutes to ensure
+ * that the token does not expire during usage.
+ * </p>
+ */
+@Log4j2
+public class GitHubClientManager {
+
+  /**
+   * The personal access token (PAT) for GitHub authentication.
+   * Note: All PAT related features should be removed in later iterations.
+   */
+  private final String ghAuthToken;
+  /**
+   * The organization name to find the installation ID of the GitHub App.
+   */
+  private final String organizationName;
+
+  /**
+   * The name of the GitHub App with the bot suffix.
+   * Example: {@code heliosapp-testing[bot]}
+   */
+  private final String appName;
+
+  /**
+   * The GitHub App ID.
+   */
+  private final Long appId;
+  /**
+   * The installation ID of the GitHub App. It can be found in organization settings.
+   * Example URL: {@code https://github.com/organizations/<org-name>/settings/installations/<installation-id>}
+   * If installationId is not provided, it will be found by the organization name.
+   */
+  private Long installationId;
+  /**
+   * The path to the private key file (pem) of the GitHub App.
+   * The private key must be generated in GitHub App settings.
+   * Generated key is in PKCS#1 format. It must be converted to PKCS#8 format.
+   */
+  private final String privateKeyPath;
+
+  private final OkHttpClient okHttpClient;
+
+  /**
+   * The GitHub client instance.
+   */
+  private volatile GitHub gitHubClient;
+  /**
+   * The current token used for authentication.
+   */
+  private volatile String currentToken;
+  /**
+   * The expiration time of the current token.
+   */
+  private volatile Instant tokenExpirationTime;
+
+  private final ReentrantLock lock = new ReentrantLock();
+
+  @Value("${http.cache.ttl}")
```

Comment:
please set a default value when no env value is specified - if possible. This makes it easier for e2e/integration test setup later on :)

I saw you added defaults in the application.yaml files, but I guess it is better to handle it here globally